.github/workflows/deploy.yml

```diff
@@ -1,32 +1,40 @@
 name: Deploy to PyPI
 
 on:
+  workflow_dispatch:
   release:
-    types: [created]
+    types:
+    - published
 
 jobs:
-  deploy:
+  dist:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Build wheel and SDist
+      run: pipx run build
+
+    - name: Check metadata
+      run: pipx run twine check dist/*
 
+    - uses: actions/upload-artifact@v2
+      with:
+        path: dist/*
+
+
+  publish:
+    needs: [dist]
     runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'published'
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Set up Python
-        uses: actions/setup-python@v1
-        with:
-          python-version: '3.x'
-
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install setuptools wheel twine
-
-      - name: Build and publish
-        env:
-          TWINE_USERNAME: ${{ secrets.pypi_username }}
-          TWINE_PASSWORD: ${{ secrets.pypi_password }}
-        run: |
-          python setup.py sdist
-          python setup.py bdist_wheel --universal
-          twine upload dist/*
+    - uses: actions/download-artifact@v2
+      with:
+        name: artifact
+        path: dist
+
+    - uses: pypa/gh-action-pypi-publish@v1.4.2
+      with:
+        user: ${{ secrets.pypi_username }}
+        password: ${{ secrets.pypi_password }}
```

Comment:
I see this in https://scikit-hep.org/developer/gha_wheels#publishing

I just looked up my management page for Uproot on PyPI. I'm the sole owner. Is the right way forward to add "scikit-hep" as an owner, and this user has a token? Presumably the password will need to change, but that's a secret that has been installed somewhere in GitHub.

Unless you recommend otherwise, I'll add "scikit-hep" as an owner. After that, is there a way to point GitHub to scikit-hep's token password?

Reply:
See https://github.com/scikit-hep/histoprint/pull/52 - basically yes, though we make scoped token for each project.

Reply:
I've made invitations for scikit-hep to co-own Uproot and Awkward Array. I'm also okay with you generating a token and replacing the secrets with that. Soon thereafter, we'll have to do a test-deployment. I'll start making a draft release.

Reply:
https://github.com/scikit-hep/uproot4/releases/tag/untagged-5ea6c0581ff469b1becf

Reply:
New secret going in, old secret going out.

Reply:
Okay, new secret in, it won't deploy unless this PR is merged. (or you remove or change the username to `__token__`)

Reply:
I'll wait on Awkward until I refactor the build-system (which I probably will do soon).

Reply:
I'll merge this now, though the first full test will be in the 4.0.8 release: https://github.com/scikit-hep/uproot4/releases/tag/untagged-8af910a834339e28bd68 (non-broken link).

I'm waiting to hear from @chrisburr on whether we should include #358 in that release.

Thanks for updating this!

Reply:
Yes it should be as xrootd 5.2 was deployed earlier today. I’ll update #358 tomorrow morning to raise a future warning.

Reply:
@chrisburr Actually, I already did, see https://github.com/scikit-hep/uproot4/pull/358/commits/d0e16b7cc63d6d3afd28c94cffaae2c92649dee8 .

Reply:
It worked! [Uproot 4.0.8](https://pypi.org/project/uproot/) uploaded with no permissions issues.